src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamConnectedStreamConformanceTests.cs

```diff
@@ -22,6 +22,25 @@ public sealed class MsQuicQuicStreamConformanceTests : QuicStreamConformanceTest
         protected override QuicImplementationProvider Provider => QuicImplementationProviders.MsQuic;
     }
 
+    public abstract class ManagedQuicStreamConformanceTestsBase : QuicStreamConformanceTests
+    {
+        protected override bool FlushRequiredToWriteData => true;
+        protected override bool FlushGuaranteesAllDataWritten => false;
```

Comment:
Is this actually the case? What does Flush guarantee then, if it doesn't guarantee that all data is written?

Reply:
To clarify the use of this flag: It's intended for something like a compression stream, where it may not be possible to actually write out all the provided input data because you've got a partial byte that can't be immediately flushed.

I don't think we should be using it for QuicStream.

Reply:
> What does Flush guarantee then, if it doesn't guarantee that all data is written?

For reference, the only two places we currently set this to false are when testing the compression streams (e.g. DeflateStream) and TranscodingStream, where there may be partial data left in the internal buffer (e.g. half a UTF8 character) that can't be written out until more data arrives to complete what's in the buffer.  I'm also surprised to see it set to false here.

Reply:
Maybe I misunderstood that one, or even what Flush() should do. Currently, flush makes the data available to the background processing thread which will send them when possible, so technically when flush returns the other side did not necessarily receive the data yet.

Reply:
With zero additional interaction with the Stream, does the Flush call guarantee that all data previously written to the Stream will eventually reach and be readable by the other end of the connection?

Reply:
> Currently, flush makes the data available to the background processing thread which will send them when possible, so technically when flush returns the other side did not necessarily receive the data yet.

Yes, that's the correct behavior for Flush, and you should be able to set FlushGuaranteesAllDataWritten to true.